mscore/pianoview.cpp

```diff
@@ -535,10 +535,14 @@ void PianoView::updateNotes()
       SegmentType st = SegmentType::ChordRest;
       for (Segment* s = staff->score()->firstSegment(st); s; s = s->next1(st)) {
             for (int track = startTrack; track < endTrack; ++track) {
-                  Chord* chord = static_cast<Chord*>(s->element(track));
-                  if (chord == 0 || chord->type() != ElementType::CHORD)
-                        continue;
-                  addChord(chord);
+                  if (s->element(track)) {
+                        if (s->element(track)->isRest())
+                              continue;
+                        Chord* chord = static_cast<Chord*>(s->element(track));
```

Comment:
Good catch but 

* Store `s->element(track)` in a temporary variable. `Element* e = s->element(track);`and use it further.
* Use e->isChord() to test that it's a Chord, instead of testing if it's not a Rest.
* use toChord(s->element(track)) instead of the static_cast.

Reply:
Also no need to test for  chord->type() != ElementType::CHORD) later.